src/core/ddsi/include/dds/ddsi/shm_transport.h

```diff
@@ -12,16 +12,17 @@
 #ifndef DDS_SHM__TRANSPORT_H
 #define DDS_SHM__TRANSPORT_H
 
-#include "dds/export.h"
-#include "iceoryx_binding_c/subscriber.h"
-#include "dds/ddsrt/sync.h"
-#include "dds/ddsi/q_protocol.h" /* for, e.g., SubmessageKind_t */
+#include "dds/ddsi/ddsi_config.h"
+#include "dds/ddsi/ddsi_keyhash.h"
 #include "dds/ddsi/ddsi_tran.h"
+#include "dds/ddsi/q_protocol.h" /* for, e.g., SubmessageKind_t */
+#include "dds/ddsrt/sync.h"
+#include "dds/export.h"
```

Comment:
I generally couldn't care less about the order of includes (which also means I don't get the point of reordering them here, or indeed in some of the other places). I know that there are some dependencies though, even when there shouldn't be. The one in particular that's scary is `dds/features.h`: I am not willing to guarantee it is really included everywhere that it matters (though I did try), so moving files ahead of that one in the order of includes requires some double-checking.

Reply:
Yes, it should not be the case that the order matters unless the headers rely on definitions to be included before the header itself is included which becomes unmaintainable fast.

The reorder is due to clang format, I must be careful with my clang format which reorders headers in lexicographic order if they are in the same block without empty lines. (ideally there should be a project specific format file).

Reply:
That explains! ☺️ 

I've never been able to convince clang-format (nor uncrustify, for that matter) to do it the way I like, but if there's a way, it would definitely be good to at least have a project-specific file in the repo.